diff --git a/bean.module b/bean.module
index f2c9bf0..1775af8 100644
--- a/bean.module
+++ b/bean.module
@@ -117,7 +117,7 @@ function bean_menu() {
 
   $items['block/%bean_delta'] = array(
     'title' => 'Block',
-    'page callback' => 'bean_view',
+    'page callback' => 'bean_view_page',
     'page arguments' => array(1),
     'access arguments' => array('view bean page'),
     'file' => 'includes/bean.pages.inc',
@@ -771,8 +771,6 @@ function bean_view(Bean $bean, $view_mode = 'default', $langcode = NULL) {
     $langcode = $GLOBALS['language_content']->language;
   }
 
-  $title = $bean->label() ? $bean->label() : $bean->adminTitle();
-  drupal_set_title($title);
   return $bean->view($view_mode, $langcode);
 }
 
diff --git a/includes/bean.pages.inc b/includes/bean.pages.inc
index 60f6f33..6cc641a 100644
--- a/includes/bean.pages.inc
+++ b/includes/bean.pages.inc
@@ -6,6 +6,15 @@
  */
 
 /**
+ * Bean view page
+ */
+function bean_view_page(Bean $bean) {
+  $title = $bean->label() ? $bean->label() : $bean->adminTitle();
+  drupal_set_title($title);
+  return bean_view($bean);
+}
+
+/**
  * Edit bean page callback
  */
 function bean_edit(Bean $bean) {
