diff --git a/bean.info b/bean.info
index bbcb4fd..d95119e 100644
--- a/bean.info
+++ b/bean.info
@@ -7,7 +7,6 @@ files[] = plugins/base.inc
 files[] = includes/translation.handler.bean.inc
 files[] = views/views_handler_filter_bean_type.inc
 files[] = bean.test
-dependencies[] = block
 dependencies[] = entity
 dependencies[] = ctools
 package = Bean
diff --git a/bean_all/plugins/all.inc b/bean_all/plugins/all.inc
index 5748072..ef79392 100644
--- a/bean_all/plugins/all.inc
+++ b/bean_all/plugins/all.inc
@@ -29,6 +29,8 @@ class BeanAll extends BeanPlugin {
     $block_object->title = $block_object->info;
     $block_object->enabled = $block_object->page_match = TRUE;
     $blocks[$this->getBlockInfo('module') . "_" . $this->getBlockInfo('delta')] = $block_object;
+
+    module_load_include('module', 'block', 'block');
     $block_content = _block_render_blocks($blocks);
     $content += _block_get_renderable_array($block_content);
 
